### DISCUSSION TITLE
Show and tell: React Zorm - Vanilla `<form>` parsing with nested fields

### DISCUSSION THREAD

**Discussion by esamattis** [colinhacks/zod#915] [Show and tell]
Hi all!

Just wanted to share this tool which allows parsing `<form>` / `<input>` to type checked nested data structures with Zod

https://github.com/esamattis/react-zorm

It works by (type-safely!) generating forms like this:

```html
<form>
    <input name="users[0].name" value="esa" />
    <input name="users[1].name" value="matti" />
</form>
```

and [parsing](https://github.com/esamattis/react-zorm/blob/78da7e2c761da970a79f74b6e5512057e60e32b3/packages/react-zorm/src/parse-form.ts) them to json like this:

```js
{
    users: [
        { name: "esa" }, 
        { name: "matti" }, 
    ]
}
```

and passing that through Zod schemas.

**Comment by JacobWeisenburger** [comment]
would you be willing to submit a PR to add this to the README in the Ecosystem section?

**Reply by esamattis** [reply]
Sure! https://github.com/colinhacks/zod/pull/1003
